includes/Modules/Analytics/Advanced_Tracking/Metadata_Collector.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Metadata_Collector
+ *
+ * @package   Google\Site_Kit\Modules\Analytics
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics\Advanced_Tracking;
+
+use WC_Product;
+
+/**
+ * Class for collecting various event metadata.
+ *
+ * @since n.e.x.t.
+ * @access private
+ * @ignore
+ */
+final class Metadata_Collector {
```

Comment:
This class appears to be entirely specific to WooCommerce, so it should be named appropriately. Also, this goes beyond the scope of this PR, let's think about this more advanced metadata gathering in a follow-up PR.

Reply:
We were planning on adding the other plugins as well, we just haven't gotten there yet.

Reply:
@j-ka11 Yeah, but this also goes a little bit beyond the initial scope I think, let's limit this PR to handling events based on data which is available in the DOM originally and that we can get with selector queries.

This class here is already complex, and adding support for the other plugins will make that more so, so I think we should move this part out to the other branch for a follow-up PR.